cmd/limactl/list.go

```diff
@@ -174,6 +179,17 @@ func listAction(cmd *cobra.Command, args []string) error {
 			}
 		}
 	}
+
+	if unmatchedInstance {
+		if len(instanceNames) > 0 {
+			err := store.PrintInstances(out, instances, format, &options)
+			if err != nil {
+				return err
+			}
+		}
+		os.Exit(1)
+	}
+
 	return store.PrintInstances(out, instances, format, &options)
```

Comment:
This looks simpler (and avoids code duplication):

```suggestion
	err := store.PrintInstances(out, instances, format, &options)
	if err == nil && unmatchedInstance {
		os.Exit(1)
	}
	return err
```

I have also a slight preference to using the plural form `unmatchedInstances` over the singular, to match the "one or more" part.